### ISSUE TITLE
Add Note Label Feature to Bookmarks

### ISSUE THREAD

**Issue by riki270601** [mihonapp/mihon#1509]
### Describe your suggested feature

Currently, there is no way to add specific labels or notes to bookmarks, making it difficult to organize and track the context or purpose of saved items.

Implement a feature that allows users to add custom notes or labels to their bookmarks. This could include:

An input field for adding a note or label when a bookmark is created or edited.

Displaying the note/label next to the bookmark for easy identification.

Allowing users to filter bookmarks by label or note.

Users can add and edit labels/notes when creating or editing a bookmark.

The labels/notes are visible alongside the bookmarks.

Users can filter or search for bookmarks based on labels or notes.

### Other details

It would be helpful for users who save many bookmarks to quickly understand the context without opening each bookmark.

A search or filter option for bookmarks based on labels would enhance usability.

This feature will improve user experience by offering better organization and quick retrieval of bookmarks based on personal context.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.17.0](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by S1332** [comment]
I have such a wide variety of genres and sources where I read from I'm desperate nowadays to add our own tags, or atleast notes cause covers and names keep changing sometimes too 🙏

**Comment by itsImprove** [comment]
+1 to this. I have bookmarks for when funny art is used, for when certain events happen in a series, for parts I like to reread etc. Adding a note or a tag to bookmarks would be super helpful.

**Comment by AntsyLich** [comment]
What happens if a source removes a chapter you added a note too?

**Comment by S1332** [comment]
When you bookmark a chapter it also carries over when you migrate sources so if it would be based on that system it should be easily possible to save the state I would imagine

**Comment by AntsyLich** [comment]
Seems like you didn't migrate after a source removed chapters. It doesn't carry over or anything.

**Comment by S1332** [comment]
Good to know. Fortunately I don't remember ever witnessing a deleted chapter

**Comment by itsImprove** [comment]
> Seems like you didn't migrate after a source removed chapters. It doesn't carry over or anything.

I've also never had a source delete a chapter they previously had, but a workaround to this would be to add a notes feature to the series itself. This way you can just note it down e.g - _Chapter XX, Nice artwork on panel XX_

This might even be better than adding notes / tags to individual chapters as you'd be able to see all the chapters bookmarked for a purpose.

**Comment by S1332** [comment]
Yeah notes and labels for entries is what I wish to use to write down what it is, I'm dyslexic af and I definitely do not Remember names that always change when my library is 1k+ filled...

> Chapter XX, Nice artwork on panel XX

I use the hold down > save image, it writes name, chapter and page Down as filename, it's fucking great

**Comment by itsImprove** [comment]
I think it's niche in people thinking about the feature, but I'd bet that a majority would appreciate and use a feature. That thread looks like a notes feature is still actively being worked on, if not finished, unless I'm reading it wrong.

**Comment by S1332** [comment]
Man that's perfect it looks good. Here's hoping it gets pushed one day.

When an source dies and i migrate I can already choose to bring my custom cover (which I use a lot) I hope it can be similar to that, cheers

**Comment by NeKoOuO** [comment]
> I think it's niche in people thinking about the feature, but I'd bet that a majority would appreciate and use a feature. That thread looks like a notes feature is still actively being worked on, if not finished, unless I'm reading it wrong.

Yeah, I personally wouldn't actively want this feature, but if it's available, I'm happy to use it.
